index.bs

```diff
@@ -816,62 +868,111 @@ To <dfn>handle funky elements</dfn> on a given |element|, run these steps:
     1. Remove the `formaction` attribute from |element|.
 </div>
 
-### The Effective Configuration ### {#configuration}
+## Matching Against The Configuration ## {#configuration}
 
-A Sanitizer is potentially complex, so we will define a helper
-construct, the *effective configuration*. This is mostly a specification
-convenience and allows us to explain a Sanitizer's operation in two steps:
-One, how to derive the effective configuration, and two, define the
-Sanitzer's operation based on it.
+A <dfn>sanitize action</dfn> can have the values `keep`, `drop`, or `block`.
 
-An effective configuration maps a given |element| or a given pair of
-|element| and |attribute| to a [=sanitize action=].
+<div algorithm="sanitize action for an element">
+To determine the <dfn>sanitize action for an |element|</dfn>, given a
+Sanitizer configuration dictionary |config|, run these steps:
 
-A <dfn>sanitize action</dfn> can have the values `keep`, `drop`, or `block`.
-To determine the <dfn>stricter action</dfn> of two [=sanitize actions=], pick
-the 'larger' of the two actions assuming a transitively defined order with
-`drop` &gt; `block`, and `block` &gt; `keep`.
+  1. Let |kind| be |element|'s [=element kind=].
+  1. If |element| is of `regular` |kind| and does not
+     [=element matches an element name|match=] any name in the
+     [=baseline element allow list=]: Return `drop`.
+  1. If |element| is of `custom` |kind| and if |config|'s
+     [=allow custom elements option=] is unset or set to anything other
+     than `true`: Return `drop`.
```

Comment:
I think `config` is a map, so we can use https://infra.spec.whatwg.org/#map-exists for marginally more precision. I think you could also simplify this by setting a default value of `false` in https://wicg.github.io/sanitizer-api/#config for both `allowCustomElements` and `allowComments`, which should remove the necessity for the exists check entirely. This would also apply to the "allow comments option" bit in 4.2 of "sanitize a node".

Reply:
Done.